vector/v.random/testsuite/test_v_random.py

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
+""" 
+Name:      v.random test
+Purpose:   Tests v.random and its flags/options.
+Authors:   Josef Pudil (original draft)
+           Sunveer Singh (finished test)
+Copyright: (C) 2020-2021 by Sunveer Singh and the GRASS Development Team
+Licence:   This program is free software under the GNU General Public
+           License (>=v2). Read the file COPYING that comes with GRASS
+           for details.
+"""
+from grass.gunittest.case import TestCase
+from grass.gunittest.main import test
+
+
+class TestVRandom(TestCase):
+    output = "test01"
+    output2 = "test02"
+    npoints = 100
+    state = "nc_state"
+    zmin = 10
+    zmax = 120
+
+    @classmethod
+    def setUpClass(cls):
+        cls.use_temp_region()
+        cls.runModule("g.region", vector=cls.state)
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.del_temp_region()
+
+    def tearDown(cls):
+        cls.runModule("g.remove", type="vector", flags="f", name=cls.output)
+        cls.runModule("g.remove", type="vector", flags="f", name=cls.output2)
+
+    # Checking if number of points equals 100
+    def test_num_points(self):
+        self.assertModule("v.random", output=self.output, npoints=self.npoints)
+
+        topology = dict(points=self.npoints)
+        self.assertVectorFitsTopoInfo(vector=self.output, reference=topology)
+
+    """Checking if the map is 3D and number of points is 100"""
+
+    def test_num_points_3D(self):
+        self.assertModule(
+            "v.random",
+            output=self.output,
+            npoints=self.npoints,
+            zmin=self.zmin,
+            zmax=self.zmax,
+            flags="z",
+        )
+
+        topology = dict(points=self.npoints, map3d=1)
+        self.assertVectorFitsTopoInfo(vector=self.output, reference=topology)
+
+    """Checking if all points are in the polygon boundary state"""
+
+    def test_restrict(self):

```

Comment:
Change of syntax is not enough here. This is not documenting the function. See how docstrings for functions work, e.g., [here](https://stackabuse.com/python-docstrings/). The result will then make more sense in the terms of formatting (empty lines, Black). Run the code but locally make some modification so that the test fails. You will the see the docstring reported with the failed tests as I mentioned earlier.